randovania/game_connection/connector/remote_connector.py

```diff
@@ -17,7 +17,8 @@
     from randovania.game_description.db.area import Area
     from randovania.game_description.db.region import Region
 
-PickupEntryWithOwner = tuple[str, PickupEntry]
+# provider_name, PickupEntry, PickupIndex, provider_uuid
+RemotePickupTuple = tuple[str, PickupEntry, PickupIndex, str]
```

Comment:
why is the uuid a string instead of uuid.UUID?

Reply:
Yeah I can't see any other usecase either. Would be fine with a boolean.

Reply:
> No hard reason.
> uuid.UUID is not json serializable. Therefore the server sends it as string and I just didn't converted it to uuid again on client side because I don't really have to.

Our automatic serialization of server/client should be handling UUID, which means it's no effort. Also a good chunk of the client-server api is actually binary, not json and so sending the UUID class is more efficient since it's just 16 bytes.

Of course it could be this part of the API uses none of that tooling, but I sure hope this rather high volume part is using it...

Reply:
I'd rather send the `uuid.UUID` and just fix the serialization tbh. should be pretty trivial to do and it's easy to believe that having the uuid might come in handy later